packages/shared/lib/core/wallet/utils/isFilteredActivity.ts

```diff
@@ -78,25 +80,39 @@ export function isFilteredActivity(activity: Activity): boolean {
         }
         if (
             filter.status.selected === StatusFilterOption.Claimed &&
-            activity.asyncStatus !== ActivityAsyncStatus.Claimed
+            activity.data.type === ActivityType.Transaction &&
+            activity.data.asyncStatus !== ActivityAsyncStatus.Claimed
         ) {
             return true
         }
         if (
             filter.status.selected === StatusFilterOption.Unclaimed &&
-            (!activity.asyncStatus || activity.asyncStatus === ActivityAsyncStatus.Claimed)
+            activity.data.type === ActivityType.Transaction &&
+            (!activity.data.asyncStatus || activity.data.asyncStatus === ActivityAsyncStatus.Claimed)
         ) {
             return true
         }
     }
     if (filter.type.active && filter.type.selected) {
-        if (filter.type.selected === TypeFilterOption.Incoming && activity.direction !== ActivityDirection.In) {
+        if (
+            filter.type.selected === TypeFilterOption.Incoming &&
+            activity.data.type === ActivityType.Transaction &&
+            activity.data.direction !== ActivityDirection.In
+        ) {
             return true
         }
-        if (filter.type.selected === TypeFilterOption.Outgoing && activity.direction !== ActivityDirection.Out) {
+        if (
+            filter.type.selected === TypeFilterOption.Outgoing &&
+            activity.data.type === ActivityType.Transaction &&
+            activity.data.direction !== ActivityDirection.Out
```

Comment:
Maybe it starts to make sense to add a helper function `isOutgoingTransaction()` and `isIncomingTransaction`? Also again it's clearer and less error prone to use `===` than `!==`.

Reply:
regarding the `===` and `!==`: i generally agree, but i think in this case `!==` makes more sense. In the filter i have the selected `FilterOption.Outgoing` therefore i want to exclude all which are not `DIrection.Out`

Reply:
But aren't all outgoing transactions ones with `Direction.out`? If you selected outgoing don't you want to have the outgoing transactions from all the transactions? Maybe I'm missing something... :laughing: